# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/php-uclibc.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- php-4.3.7/ext/standard/dns.c.orig	2003-09-02 00:37:46.000000000 +0200
+++ php-4.3.7/ext/standard/dns.c	2004-11-23 11:59:36.973971768 +0100
@@ -325,15 +325,23 @@
 	cp = (u_char *)&ans + HFIXEDSZ;
 	end = (u_char *)&ans +i;
 	for (qdc = ntohs((unsigned short)hp->qdcount); qdc--; cp += i + QFIXEDSZ) {
+#if HAVE_DN_SKIPNAME
 		if ((i = dn_skipname(cp, end)) < 0 ) {
 			RETURN_FALSE;
 		}
+#else
+		RETURN_FALSE;
+#endif
 	}
 	count = ntohs((unsigned short)hp->ancount);
 	while (--count >= 0 && cp < end) {
+#if HAVE_DN_SKIPNAME
 		if ((i = dn_skipname(cp, end)) < 0 ) {
 			RETURN_FALSE;
 		}
+#else
+		RETURN_FALSE;
+#endif
 		cp += i;
 		GETSHORT(type, cp);
 		cp += INT16SZ + INT32SZ;
